src/Common/MemoryTrackerDebugBlockerInThread.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <base/defines.h> /// DEBUG_OR_SANITIZER_BUILD
+
+#ifdef DEBUG_OR_SANITIZER_BUILD
+#include <cstdint>
+
+struct MemoryTrackerDebugBlockerInThread
+{
+public:
+    MemoryTrackerDebugBlockerInThread();
+    ~MemoryTrackerDebugBlockerInThread();
+
+    static bool isBlocked() { return counter > 0; }
+    static constexpr bool isEnabled() { return true; }
+
+private:
+    static thread_local uint64_t counter;
+};
+#else
+struct MemoryTrackerDebugBlockerInThread
+{
+public:
+    static constexpr bool isEnabled() { return false; }
+    static constexpr bool isBlocked() { return true; }
```

Comment:
```suggestion
    static constexpr bool isBlocked() { return false; }
```

Reply:
Can you elaborate?

Since in case of `!DEBUG_OR_SANITIZER_BUILD` (**no sanitizers or debug**), then it does not matter since `isEnabled` is `false` anyway, and this `isBlocked` will never be called. I used `true` just to underline that it will not be used in release builds.

Reply:
I know, I just thought it would be a little clearer if the logic was the completely opposite to the code in the debug build. But okay, let's leave it as is.